packages/@rnw-scripts/doxysaurus/src/markdown.ts

```diff
@@ -0,0 +1,549 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ *
+ * This code is partially based on moxygen and doxygen2md code.
+ * The copyrights of moxygen and doxygen2md projects:
+ * Copyright (c) 2016 Philippe FERDINAND
+ * Copyright (c) 2016 Kam Low
+ *
+ * @format
+ **/
+
+//
+// Transforms Doxygen descriptions to markdown.
+//
+
+import {DoxDescription, DoxDescriptionElement} from './doxygen-model';
+import {DocCompound, DocMemberOverload} from './doc-model';
+import {log} from './logger';
+
+export interface TypeLinks {
+  linkPrefix: string;
+  linkMap: Map<string, string>;
+  operatorMap: Map<string, string>;
+}
+
+export interface LinkResolver {
+  stdTypeLinks: TypeLinks;
+  idlTypeLinks: TypeLinks;
+  resolveCompoundId(doxCompoundId: string): DocCompound | undefined;
+  resolveMemberId(
+    doxMemberId: string,
+  ): [DocCompound | undefined, DocMemberOverload | undefined];
+}
+
+export function toMarkdown(desc: DoxDescription, linkResolver?: LinkResolver) {
+  const context: DoxDescriptionElement[] = [];
+  let indent = 0;
+
+  type AutoLinkerState = 'Start' | 'InTemplateArgs' | 'InMember';
+  let linkerState: AutoLinkerState = 'Start';
+  let linkerTemplateDepth = 0;
+  let linkerTypeLink: string | undefined;
+
+  const sb = new StringBuilder();
```

Comment:
personally not a fan of "reusing" global sb variable. 
My mental model would prefer to see it passed along...

I know this is nested functions in nested functiosn so it functionally works fine... It is a personal preference thing...

Reply:
I have started this code with TS classes. It was very difficult to read because of this `this.` boilerplate all over the place. Even the invocation was looking complex. I like to use the JavaScript language features as they allow to write compact code. The use of the 'sb' variable is not different from the lambda captures in C++. I have never heard that capturing variables in a lambda is an anti-pattern.